cmd/traffic/cmd/manager/internal/state/state.go

```diff
@@ -97,22 +97,24 @@ type State struct {
 	//     `intercept.ClientSession.SessionId`)
 	//  6. `intercepts` needs to be pruned in-sync with `agents` (based on
 	//     `agent.Name == intercept.Spec.Agent`)
-	port         uint16
-	intercepts   watchable.InterceptMap
-	agents       watchable.AgentMap                   // info for agent sessions
-	clients      watchable.ClientMap                  // info for client sessions
-	sessions     map[string]SessionState              // info for all sessions
-	listeners    map[string]connpool.Handler          // listeners for all intercepts
-	agentsByName map[string]map[string]*rpc.AgentInfo // indexed copy of `agents`
```

Comment:
Many of the contents of this commit are reverted in https://github.com/telepresenceio/telepresence/pull/1772/commits/5c9ab850915ac9c2d5d17b0dd81509a02ac1aaaf ... The reason for that was bc I originally thought we should call "RemoveIntercept" when the user left an intercept, but through later development realized we'd need to add the same thing to the garbage-collection go routine, and so thought it unnecessary to do it two times.  

That being said, I did want to leave this commit up in case y'all feel it is worth doing it twice or prefer this method to just the garbage collection.